ivy.el

```diff
@@ -2680,14 +2687,16 @@ See `completion-in-region' for further information."
            (when (eq collection 'crm--collection-fn)
              (setq comps (delete-dups comps)))
            (let* ((len (ivy-completion-common-length (car comps)))
+                  (prefix-offset (ivy--completion-prefix-offset str md))
+                  (target-str (substring str prefix-offset))
                   (initial (cond ((= len 0)
                                   "")
-                                 ((let ((str-len (length str)))
+                                 ((let ((str-len (length target-str)))
                                     (when (> len str-len)
-                                      (setq len str-len)
-                                      str)))
+                                      (setq len str-len)))
+                                  target-str)
                                  (t
-                                  (substring str (- len))))))
+                                  (substring target-str (- len))))))
```

Comment:
Do you have any examples where this branch is taken and is useful?
I'm struggling to trigger it in cases other than when `len` = `str-len`, where the `substring` is a no-op.

Reply:
Ah, I think it's for cases like completing `./foo` where the first candidate is `foobar`.  `foobar` presumably has a matching prefix of `foo`, so the last three characters of `./foo` are deleted and replaced with the entire candidate `foobar`.

Not sure if this happens outside of file name completion.